src/eth_provider/provider.rs

```diff
@@ -389,15 +388,13 @@ where
 
     async fn get_logs(&self, filter: Filter) -> EthProviderResult<FilterChanges> {
         let current_block = self.block_number().await?.try_into().map_err(|_| EthApiError::UnknownBlockNumber)?;
-        let from = filter.get_from_block().unwrap_or_default();
-        let to = filter.get_to_block().unwrap_or(current_block);
-
-        let (from, to) = match (from, to) {
-            (from, _) if from > current_block => return Ok(FilterChanges::Empty),
-            (from, to) if to > current_block => (from, current_block),
-            (from, to) if to < from => return Ok(FilterChanges::Empty),
-            _ => (from, to),
-        };
+
+        let (from, to) =
+            match (filter.get_from_block().unwrap_or_default(), filter.get_to_block().unwrap_or(current_block)) {
```

Comment:
I'm not a huge fan of this change, feel like we lose readability

Reply:
Sure, reverted :)